data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/ReactiveKafkaProducer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2018 Knative Authors (knative-dev@googlegroups.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.knative.eventing.kafka.broker.dispatcher;
+
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import io.vertx.core.Future;
+
+public interface ReactiveKafkaProducer<K, V> {
```

Comment:
Can't we use the existing interface from the receiver?

Reply:
I have also thought about that and we can do that as well, but then I thought that by doing so, the dispatcher-vertx module will also get depend on the receiver modules, and I decided not to do that.
But now we definitely have redundancy of code.
We can definitely use the existing interface.

Reply:
I would go with reducing duplication and depend on the receiver module, eventually, we can think about moving those in core module

Reply:
What's the downside of introducing a dependency on the receiver?

Reply:
No, I don't think there is any downside, but as there are no dependency already so i just thought not to introduce one.

Also, I think it will be better if we move all the interfaces to the core module.